### PR TITLE
Fix support TLS configuration from docker context

### PR DESCRIPTION
**Pull request by piotrwielgolaski-tomtom**
initial implementation was done based on docker documentation of output of CLI, which is different that real files stored by docker

### CODE REVIEW

**Comment by piotrwielgolaski-tomtom** [comment]
This is fixed for my PR https://github.com/docker-java/docker-java/pull/2105

**Comment by piotrwielgolaski-tomtom** [comment]
It is connected to my other PR in test-containers https://github.com/testcontainers/testcontainers-java/pull/7178

**Comment by piotrwielgolaski-tomtom** [comment]
@eddumelendez I noticed that my initial implementation was wrong as I based it on documentation from docker cli not looking at real files on filesystem, this time I did full check and make it work with test container, so it should be fine to merge this and connected PR in testcontainers

**Comment by piotrwielgolaski-tomtom** [comment]
some insight can be found in here https://github.com/docker/cli/blob/33c7baa7edf962559a07c1fb59581f863a7e8862/cli/context/store/store.go#L26

**Comment by piotrwielgolaski-tomtom** [comment]
I don't know what kind of description you have in mind but let me summarised what happen.

Initial support for docker context was lacking support for TLS, so it was not working in case you want to use docker remote with TLS - this was initial support https://github.com/docker-java/docker-java/pull/2036

In attempt to extend it I created PR https://github.com/docker-java/docker-java/pull/2105, but I assumed that was is described as output of `docker context inspect ` is in fact in files on disk.
I assumed that `Storage` section exists in file, but after you merged it and I tried to use it I noticed that in fact this information is only gather in runtime by docker cli which can be track to https://github.com/docker/cli/blob/33c7baa7edf962559a07c1fb59581f863a7e8862/cli/context/store/store.go#L26

In my next attempt (this PR) I dig dipper what in fact is stored in  ~/.docker/contexts and there are two folders for metadata and for tls. 
```
ls -1 ~/.docker/contexts
meta
tls
```
This PR is relying on folder structure to pick up TLS information and provide them to docker client when context is used.

I hope it is enough, but if not, let me know what more I can do to make this merged

File: docker-java-core/src/main/java/com/github/dockerjava/core/DockerContextMetaFile.java [hunk 1]
```diff
@@ -52,6 +43,16 @@ public static Optional<DockerContextMetaFile> resolveContextMetaFile(ObjectMappe
...
 
+    public static Optional<File> resolveContextTLSFile(File dockerConfigPath, String context) {
+        final File path = dockerConfigPath.toPath()
+            .resolve("contexts")
+            .resolve("tls")
+            .resolve(metaHashFunction.hashString(context, StandardCharsets.UTF_8).toString())
+            .resolve("docker")
+            .toFile();
+        return  Optional.ofNullable(path).filter(File::exists);
+    }
```
Comment (eddumelendez) [inline docker-java-core/src/main/java/com/github/dockerjava/core/DockerContextMetaFile.java:54]: What I can see by doing `docker context inspect desktop-linux` is

```
[
    {
        "Name": "desktop-linux",
        "Metadata": {
            "Description": "Docker Desktop"
        },
        "Endpoints": {
            "docker": {
                "Host": "unix:///Users/eddumelendez/.docker/run/docker.sock",
                "SkipTLSVerify": false
            }
        },
        "TLSMaterial": {},
        "Storage": {
            "MetadataPath": "/Users/eddumelendez/.docker/contexts/meta/fe9c6bd7a66301f49ca9b6a70b217107cd1284598bfc254700c989b916da791e",
            "TLSPath": "/Users/eddumelendez/.docker/contexts/tls/fe9c6bd7a66301f49ca9b6a70b217107cd1284598bfc254700c989b916da791e"
        }
    }
]
```

and the tls can be retrieved using the `Storage > TLSPath` in the json file, plus adding the `docker` folder at the end, right? or is there any other motivation behind this change?

**Comment by eddumelendez** [comment]
Thanks! Indeed, it helps. I wanted to clarify this before to continue.

File: docker-java-core/src/main/java/com/github/dockerjava/core/DockerContextMetaFile.java (hunk 1 above)
Comment (piotrwielgolaski-tomtom) [reply docker-java-core/src/main/java/com/github/dockerjava/core/DockerContextMetaFile.java:54]: Try to find it on disk, as I wrote this output is composed by docker cli, on this there is folder, but this information is not in json file - it was my initial idea, which was proven to not work, so there is this PR to fix it

Comment (eddumelendez) [reply docker-java-core/src/main/java/com/github/dockerjava/core/DockerContextMetaFile.java:54]: got it! I forgot that this is not stored 👍🏽

I’ll take another look later today and merge it.

Thanks!

**Review by eddumelendez — APPROVED** [review]
For further reference, I tested it following https://blog.devgenius.io/docker-security-securing-docker-daemon-using-tls-5234b4782047

**Comment by eddumelendez** [comment]
Thanks for your contribution, @piotrwielgolaski-tomtom !

**Pull request merged**